doc-content/shared-kie-docs/src/main/asciidoc/KieServer/kie-server-commands-samples-ref.adoc

```diff
@@ -0,0 +1,876 @@
+              "scanner": {
+                "status": "STARTED",
+                "poll-interval": 5000
+              },
+              "config-items": [
+                {
+                  "itemName": "RuntimeStrategy",
+                  "itemValue": "SINGLETON",
+                  "itemType": "java.lang.String"
+                },
+                {
+                  "itemName": "MergeMode",
+                  "itemValue": "MERGE_COLLECTIONS",
+                  "itemType": "java.lang.String"
+                },
+                {
+                  "itemName": "KBase",
+                  "itemValue": "",
+                  "itemType": "java.lang.String"
+                },
+                {
+                  "itemName": "KSession",
+                  "itemValue": "",
+                  "itemType": "java.lang.String"
+                }
+              ],
+              "messages": [
+                {
+                  "severity": "INFO",
+                  "timestamp": {
+                    "java.util.Date": 1538504619749
+                  },
+                  "content": [
+                    "Container command-script-container successfully created with module com.redhat:Project1:1.0."
+                  ]
+                }
+              ],
+              "container-alias": null
+            }
+          ]
+        }
+      }
+    }
+  ]
+}
+----
+--
+CallContainerCommand::
++
+--
+Calls a KIE container and executes one or more runtime commands. For information about {PRODUCT} runtime commands, see xref:runtime-commands-con_kie-apis[].
+
+.Command attributes
+[cols="30%,50%,20%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Requirement
+
+|`containerId`
+|ID of the KIE container to be called
+|Required
+
+|`payload`
+|One or more commands in a `BatchExecutionCommand` wrapper to be executed on the KIE container
+|Required
+
+|
+|
+|Optional
+|===
+
+*<@Maciej/@Tiho: The above attributes table is missing the rest of the required and optional parameters, and the below example Java command contains only the required payload portion and needs the rest of the call-container command to be complete.>*
+
+.Example Java command
+[source,java]
+----
+import org.kie.server.api.marshalling.Marshaller;
+import org.kie.server.api.marshalling.MarshallerFactory;
+import org.kie.server.api.marshalling.MarshallingFormat;
+
+...
+
+Marshaller marshaller = MarshallerFactory.getMarshaller(MarshallingFormat.JSON,myclass.class.getClassLoader());
+//Marshalling format is changed based on the method of marshalling for the CallContainerCommand. Also note myclass.class classloader is called. If replicating this code, change the name to the name of your class.
+
+Command<?> fire = KieServices.Factory.get().getCommands().newFireAllRules();
+BatchExecutionCommand batch = KieServices.Factory.get().getCommands().newBatchExecution(Arrays.<Command<?>>asList(fire), "defaultKieSession");
+String payload = marshaller.marshall(batch);
```

Comment:
users should not use this approach, instead they should simply rely on RuleServicesClient.executeCommandsWithResults which accepts command - in most of the cases BatchExecutionCommand and it will do all the marshalling for you

````
List<Command<?>> commands = new ArrayList<Command<?>>();
        BatchExecutionCommand batchExecution1 = commandsFactory.newBatchExecution(commands, KIE_SESSION_1);

        commands.add(commandsFactory.newInsert(person, PERSON_OUT_IDENTIFIER));
        commands.add(commandsFactory.newFireAllRules());

        ServiceResponse<ExecutionResults> response1 = ruleClient.executeCommandsWithResults(CONTAINER_ALIAS, batchExecution1);
       

````

Reply:
Done.